compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_vector_distribution.mlir

```diff
@@ -305,94 +305,127 @@ builtin.module attributes { transform.with_named_sequence } {
-  // CHECK:        %[[D12:.+]] = vector.bitcast %[[D11]] : vector<2xf16> to vector<1xi32>
-  // CHECK:        %[[D13:.+]] = vector.extract %[[D12]][0] : i32 from vector<1xi32>
-  // CHECK:        %[[SHUFFLERESULT:.+]], %[[VALID:.+]] = gpu.shuffle  xor %[[D13]], %[[C16_I32]], %[[C32_I32]] : i32
-  // CHECK:        %[[D14:.+]] = vector.broadcast %[[SHUFFLERESULT]] : i32 to vector<1xi32>
-  // CHECK:        %[[D15:.+]] = vector.bitcast %[[D14]] : vector<1xi32> to vector<2xf16>
-  // CHECK:        %[[D16:.+]] = arith.maximumf %[[D15]], %[[D11]] : vector<2xf16>
-  // CHECK:        %[[D17:.+]] = vector.bitcast %[[D16]] : vector<2xf16> to vector<1xi32>
-  // CHECK:        %[[D18:.+]] = vector.extract %[[D17]][0] : i32 from vector<1xi32>
-  // CHECK:        %[[SHUFFLERESULT_1:.+]], %[[VALID_2:.+]] = gpu.shuffle  xor %[[D18]], %[[C32_I32]], %[[C32_I32]] : i32
-  // CHECK:        %[[D19:.+]] = vector.broadcast %[[SHUFFLERESULT_1]] : i32 to vector<1xi32>
-  // CHECK:        %[[D20:.+]] = vector.bitcast %[[D19]] : vector<1xi32> to vector<2xf16>
-  // CHECK:        %[[D21:.+]] = arith.maximumf %[[D20]], %[[D16]] : vector<2xf16>
-  // CHECK:        %[[D22:.+]] = vector.extract %[[D21]][0] : f16 from vector<2xf16>
-  // CHECK:        %[[D23:.+]] = vector.extract %[[D21]][1] : f16 from vector<2xf16>
-  // CHECK:        %[[D24:.+]] = arith.maximumf %[[D22]], %[[D23]] : f16
-  // CHECK:        %[[D25:.+]] = arith.maximumf %[[D24]], %[[D1]] : f16
-  // CHECK:        %[[D26:.+]] = vector.insert %[[D25]], %[[CST_0]] [0] : f16 into vector<1xf16>
-  // CHECK:        %[[D27:.+]] = iree_vector_ext.to_simd %[[D26]] : vector<1xf16> -> vector<16xf16>
-  %result = vector.multi_reduction <maximumf>, %source, %init {
-                "__vector_layout_test_anchor_operand_0" = #layout2d,
-                "__vector_layout_test_anchor_operand_1" = #layout1d,
-                "__vector_layout_test_anchor_result_0" = #layout1d
-              } [0] : vector<16x16xf16> to vector<16xf16>
-  func.return %result : vector<16xf16>
+// Dummy hal executable ops to set the subgroup size which is required for distributing reductions.
+#executable_target_rocm_hsaco_fb = #hal.executable.target<"rocm", "rocm-hsaco-fb", {}>
+#pipeline_layout = #hal.pipeline.layout<push_constants = 0, sets = [<0, bindings = [<0, storage_buffer, ReadOnly>]>]>
+hal.executable private @reduction_dispatch {
+  hal.executable.variant public @rocm_hsaco_fb target(#executable_target_rocm_hsaco_fb) {
+    hal.executable.export public @distribute_reduction_f16 ordinal(0) layout(#pipeline_layout) attributes {subgroup_size = 64 : index} {
+      ^bb0(%arg0: !hal.device):
+        %c2 = arith.constant 2 : index
+        %c32 = arith.constant 32 : index
+        %c1 = arith.constant 1 : index
+        hal.return %c2, %c32, %c1 : index, index, index
+    }
+    builtin.module {
+      func.func @distribute_reduction_f16(%source: vector<16x16xf16>, %init: vector<16xf16>) -> vector<16xf16> {
```

Comment:
@MaheshRavishankar I know you're in the middle of trying to squash inclusion of variants in tests, should we try to add the subgroup size annotation to the function in the short term (for testing)?

Reply:
Happy to query information from functions instead of adding dummy hal ops if that's a possibility.